CRM/Utils/File.php

```diff
@@ -1138,7 +1138,7 @@ public static function isDir(?string $dir) {
       return FALSE;
     });
     try {
-      $is_dir = is_dir($dir);
+      $is_dir = is_dir($dir ?? '');
```

Comment:
For this one I'd like to check what they did upstream in php itself. For security reasons this function should behave identical to is_dir with some small exceptions.

Reply:
So it looks like php itself does not have a test for null on this. My worry is that there is some edge-case situation where the empty string would return true even though you passed in null, bypassing security. I'm thinking the answer is to update the test to expect something different based on php version, since that is what is actually the case.

I'll take a stab at that.

Reply:
This is tricky because it's a deprecation which is hard to get at. How about for now just before the `try` on line 1140:

```php
if ($dir === NULL) {
  return FALSE;
}
```

instead of the `?? ''`

Reply:
Or no wait it would have to be before the set_error_handler - the very first thing in the function.

Reply:
Updated